src/main/resources/org/opentripplanner/apis/transmodel/schema.graphql

```diff
@@ -1215,6 +1215,36 @@ type TimeAndDayOffset {
   time: Time
 }
 
+"""
+The time-penalty is applied to either the access-legs and/or egress-legs. Both access and
+egress may contain more than one leg; Hence, the penalty is not a field on leg.
+
+Note! This is for debugging only. This type can change without notice.
+"""
+type TimePenalty {
```

Comment:
TimePenalty having a cost and a time penalty is weirdly recursive.

Reply:
In the RouteRequest this concept seems to be called simply "penalty" under `accessEgress` if I understand correctly? And in the source code it's mostly called TimeAndCostPenalty. Would one of those names make sense instead?

Reply:
Naming is hard. The reason why I struggle with this one is that it is the penalty on time witch is the main thing looking at this from a use-case perspective. Propagating a penalty on time to cost is secondary. We could apply the penalty to time only, and not to cost, adjusting the cost mode reluctance instead. But, since cost is highly dependent on time it feels natural to do it. All parts that take time in a journey has a reluctance and propagate to `generalized-cost`. Calling this TimeAndCostPenalty is misleading - yes it affect both, but the ONLY reason to use it is to apply the penalty on time. `TimePenaltyThatPropagatesToCost` is a bit long ;-) 

So maybe `TimePenaltyWithCost` is ok? It at least indicate that cost is secondary.

Should I rename this and the internal type? The `AccessEgressPreferences.penalty` should probably be `timePenalty` - it is a field and the name should not be to long and it should reflect the purpouse, not describe the content.